game-core/src/main/java/games/strategy/triplea/ui/mapdata/PlayerColors.java

```diff
@@ -48,7 +48,14 @@ Color getPlayerColor(final String playerName) {
         "Illegal player name: %s, use the method 'getImpassableColor()' instead",
         playerName);
 
-    return playerColors.computeIfAbsent(playerName, this::computePlayerColor);
+    // NOTE: we do *not* use computeIfAbsent here to avoid
+    // 'java.util.ConcurrentModificationException'
+    Color playerColor = playerColors.get(playerName);
+    if (playerColor == null) {
+      playerColor = computePlayerColor(playerName);
+      playerColors.put(playerName, playerColor);
+    }
+    return playerColor;
```

Comment:
Feel free to correct me if I'm wrong but isn't a `ConcurrentModificationException` a more correct behaviour?
If for example 2 Threads try to query the player name at the same time, with this code no exception would be thrown, but it could happen that the code computes 2 different colors, which are both returned, but only one of them is stored in the map.
Unlikely scenario, but we might want to consider using a synchronized/concurrent Map instead

Reply:
This is an example of resolving the race condition by 'let it happen'.
In this case, the update is to the same value, we guaranteed that player color computation is deterministic.

Reply:
Let's also consider if we were to resolve the race condition via exception and that the colors were different. In this case one writer could write a value, the value then read by another reader. The other writer would come in and set the value to something else, meanwhile the other reader has the old value, and that would be incorrect. Having a concurrent exception does not solve the race condition as we need to be sure that any readers are going to get consistent results. So this is properly solved with deterministic player color computation (nonetheless it getting written multiple times is not clean).